models/article-models.js

```diff
@@ -2,22 +2,35 @@ const db = require("../db/connection");
 
 exports.fetchArticleById = (article_id) => {
   return db
-    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
-    .then((article) => {
-      if (article.rows.length === 1) {
-        return article.rows[0];
-      } else {
+    .query(
+      "SELECT articles.article_id, articles.author, title, topic, articles.body, articles.created_at, articles.votes FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id WHERE articles.article_id = $1",
+      [article_id]
+    )
+    .then((results) => {
+      if (results.rows.length === 0) {
         return Promise.reject({ status: 404, msg: "article not found" });
+      } else {
+        let count = 0;
+        const article = results.rows[0];
+        article["comment_count"] = count;
+        if (results.rows.length > 1) {
+          article.comment_count = results.rows.length;
+          return article;
+        } else {
+          return article;
+        }
```

Comment:
Ideally, you need to solve this comment_count issue with a SQL query to show off your sql skills. 

I'd look at aggregate functions such as COUNT and then group_by.